queue_job/job.py

```diff
@@ -238,6 +238,31 @@ def load_many(cls, env, job_uuids):
         recordset = cls.db_records_from_uuids(env, job_uuids)
         return {cls._load_from_db_record(record) for record in recordset}
 
+    def lock(self):
+        self.env.cr.execute(
+            """
+            SELECT
+                *
+            FROM
+                queue_job_locks
+            WHERE
+                id in (
+                    SELECT
+                        id
+                    FROM
+                        queue_job
+                    WHERE
+                        uuid = %s
+                        AND state='started'
+                )
+            FOR UPDATE;
+        """,
+            [self.uuid],
+        )
```

Comment:
```suggestion
            WHERE
                id = %s
            FOR UPDATE;
        """,
            [self.id],
        )
```

Since the job lock id is the job id?

Reply:
The job lock id is the job id.
But, I need to ensure that the job is `started`.
So, I'll keep my query as it is.